x-pack/functionbeat/config/config.go

```diff
@@ -134,3 +138,40 @@ func (p *ProviderConfig) Validate() error {
 	}
 	return nil
 }
+
+// MemSizeFactor64 implements a human understandable format for bytes but also make sure that all
+// values used are a factory of 64.
+type MemSizeFactor64 int
```

Comment:
This type looks quite specific. It is used to unpack a setting that is supposed to be a multiple of 64MB? Would it be better to only introduce the type where it is needed?

Reply:
It is the requirement by Cloudformation. However, this requirement does not interfere with the GCP settings, to I took the liberty to use it for both providers. It does not need much work to move it back to the provider-specific config if it would lead to a clearer architecture.